cdci_data_analysis/flask_app/app.py

```diff
@@ -277,9 +277,34 @@ def free_up_space():
 
 @app.route('/inspect-state', methods=['POST', 'GET'])
 def inspect_state():
+    token = request.args.get('token', None)
+
+    app_config = app.config.get('conf')
+    secret_key = app_config.secret_key
+    output, output_code = tokenHelper.validate_token_from_request(token=token, secret_key=secret_key,
+                                                                  required_roles=['job manager'],
+                                                                  action="inspect the state for a given job_id")
+
+    if output_code is not None:
+        return make_response(output, output_code)
     state_data_obj = InstrumentQueryBackEnd.inspect_state(app)
-    return state_data_obj
+    return jsonify(dict(records=state_data_obj['records']))
+
 
+@app.route('/inspect-jobs', methods=['POST', 'GET'])
```

Comment:
could you just make this part of the `inspect-state` with some extra argument?

Reply:
I thought about, but they serve two purposes that are quite different, and the relative outputs are quite different. 

Reason why I preferred to have two separate endpoints...

Reply:
Why different, what are the purposes?

They are not called in meaningfully different ways, both return dispatcher state which consists of set of jobs.

Reply:
`inspect-jobs` returns a list of objects, each related to a specific job. It serves the purpose of reporting the status of a job, by reading the content of the various scratch_dir related to that job, and understand whether or not that job has completed

`inspect-state`returns a list of objects, each containing information about the content of a single scratch_dir, and multiple scratch_dir might be related to the same job

perhaps more appropriate names should be assigned...

Reply:
They both return information about jobs (derived from scratch directories), just grouped differently, and with more details in the new case. `inspect-jobs` in just further development on `inspect-state` and I still do not see why it has to be separate.

To maintain previous interface, we can have an option to return previous-style structure.

Reply:
but they both contain information that serve different purposes, and they should be both maintained (perhaps with a more appropriate naming):

- `inspect-state` contains info regarding the email_history, matrix_message_history, session_log, fits_files etc for a single scratch_dir
- `inspect-jobs` contains info "only" related to the job itself: in particular, its status and it also includes the query_output (extracted from the `query_output.json` file), plus, a logic is implemented to understand whether or not that job has completed

So effectively, the information returned from one are not contained into the other (and viceversa).

That is why I think inspect-jobs cannot be seen as an extension of inspect-state

Reply:
They contain information serving the same purpose: inspecting state record of the job history, for the purpose of assessing what messages were received and sent in relation to each job. Do they not?

Reply:
With inspect-state the focus is on the content of the single scratch folder.

With inspect-jobs the focus is on the single job.

I see this as the main difference, besides the returned content itself.

In any case, they can be "merged" so that inspect-jobs also includes  what inspect-state returns

Reply:
This sort of focus difference is just difference in rearranging output.

Job info record is state record. So I don't see why it's needed to renamed the endpoint.
Just add a request argument to `inspect-state` something like `group_by_job`. Also maybe other arguments would be useful to reduce the amount of files open, given the test it might end up being some 10x1000 files in production.

Reply:
but then we should include also what inspect-state already does (email_history etc etc)...

Reply:
included now, and json schema adapted

Reply:
to reduce the number of opened files, I added also the arg `exclude_analysis_parameters` that if set, would exclude the content from the `analysis_parameters.json` file

by adding this the `token_expired` won't be present as it cannot be evaluated